dali/core/mm/default_resources.cc

```diff
@@ -163,6 +159,50 @@ bool UseVMM() {
   return value;
 }
 
+size_t MallocThreshold() {
+  char *env = getenv("DALI_MALLOC_POOL_THRESHOLD");
+  int len = 0;
+  if (env && (len = strlen(env))) {
+    for (int i = 0; i < len; i++) {
+      bool valid = std::isdigit(env[i]) || (i == len - 1 && (env[i] == 'k' || env[i] == 'M'));
+      if (!valid) {
+        DALI_FAIL(make_string(
+          "DALI_MALLOC_POOL_THRESHOLD must be a number, optionally followed by 'k' or 'M', got: ",
+          env));
+      }
+    }
+    size_t s = atoll(env);
+    if (env[len-1] == 'k')
+      s <<= 10;
+    else if (env[len-1] == 'M')
+      s <<= 20;
+    return s;
+  } else {
+    char *mmap = getenv("MALLOC_MMAP_MAX_");
+    if (mmap && !atoi(mmap))
+      return 0;
+    char *thresh = getenv("MALLOC_MMAP_THRESHOLD_");
+    if (thresh)
+      return atoll(thresh);
+    return (32 << 20);  // max for 64-bit Linux systems
+  }
+}
+
+inline std::shared_ptr<host_memory_resource> CreateDefaultHostResource() {
+  static auto rsrc = std::make_shared<malloc_memory_resource>();
+  size_t threshold = MallocThreshold();
+  if (threshold > 0) {
```

Comment:
Maybe we should have a default value for that?

Reply:
We do - it's in MallocThreshold.

Reply:
Right, missed that.